pkg/apis/local/v1alpha1/localvolumediscoveryresult_types.go

```diff
@@ -0,0 +1,95 @@
+package v1alpha1
+
+import (
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// DeviceState defines the observed state of the disk
+type DeviceState string
+
+const (
+	// Available means that the device is available to use and a new persistent volume can be provisioned on it
+	Available DeviceState = "Available"
+	// NotAvailable means that the device is already used by some other process and shouldn't be used to provision a Persistent Volume
+	NotAvailable DeviceState = "NotAvailable"
+	// Unknown means that the state of the device can't be determined
+	Unknown DeviceState = "Unknown"
+)
+
+// DeviceStatus defines the observed state of the discovered devices
+type DeviceStatus struct {
+	// State shows the availability of the device
+	State DeviceState `json:"state"`
+}
+
+// DiscoveredDevice shows the list of discovered devices with their properties
+type DiscoveredDevice struct {
+	// DeviceID represents the persistent name of the device. For eg, /dev/disk/by-id/...
+	DeviceID string `json:"deviceID"`
+	// Path represents the device path. For eg, /dev/sdb
+	Path string `json:"path"`
+	// Model of the discovered device
+	Model string `json:"model"`
+	// Type of the discovered device
+	Type DeviceType `json:"type"`
+	// Vendor of the discovered device
+	Vendor string `json:"vendor"`
+	// Serial number of the disk
+	Serial string `json:"serial"`
+	// Size of the discovered device
+	Size resource.Quantity `json:"size"`
+	// Property represents whether the device type is rotational or not
+	Property DeviceMechanicalProperty `json:"property"`
+	// FSType represents the filesystem available on the device
+	FSType string `json:"fstype"`
+	// Status defines whether the device is available for use or not
+	Status DeviceStatus `json:"status"`
```

Comment:
this basically means everything is required here. But for many of these fields no information may be available. For example - `fsType` may be only available if there is a filesystem on the disk and in which case what happens?